walker.go

```diff
@@ -0,0 +1,57 @@
+package chi
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type WalkFn func(method string, route string, handler http.Handler, middlewares ...func(http.Handler) http.Handler) error
+
+func Walk(r Routes, fn WalkFn) error {
```

Comment:
Cool - btw, I didn't get my head so deep to trace this logic - can you tell me why this walk method is better then just using the chi.Routes interface on a router? just like docgen does.. is there some extra data you're able to reach into?